advanced_functionality/autogluon-sagemaker-pipeline/pipelines/abalone/pipeline.py

```diff
@@ -0,0 +1,292 @@
+"""Example workflow pipeline script for abalone pipeline.
+
+                                               . -RegisterModel
+                                              .
+    Process-> Train -> Evaluate -> Condition .
+                                              .
+                                               . -(stop)
+
+Implements a get_pipeline(**kwargs) method.
+"""
+import os
+
+import boto3
+import sagemaker
+import sagemaker.session
+
+from sagemaker.estimator import Estimator
+from sagemaker.inputs import TrainingInput
+from sagemaker.model_metrics import (
+    MetricsSource,
+    ModelMetrics,
+)
+from sagemaker.processing import (
+    ProcessingInput,
+    ProcessingOutput,
+    ScriptProcessor,
+)
+from sagemaker.sklearn.processing import SKLearnProcessor
+from sagemaker.workflow.conditions import ConditionLessThanOrEqualTo
+from sagemaker.workflow.condition_step import (
+    ConditionStep,
+    JsonGet,
+)
+from sagemaker.workflow.parameters import (
+    ParameterInteger,
+    ParameterString,
+)
+from sagemaker.workflow.pipeline import Pipeline
+from sagemaker.workflow.properties import PropertyFile
+from sagemaker.workflow.steps import (
+    ProcessingStep,
+    TrainingStep,
+)
+from sagemaker.workflow.step_collections import RegisterModel
+from pipelines.ag_model import AutoGluonTraining, AutoGluonTabularPredictor, AutoGluonInferenceModel
+
+# official autogluon images
+image_uri = (
+    "763104351884.dkr.ecr.us-east-1.amazonaws.com/autogluon-training:0.4.2-cpu-py38-ubuntu20.04"
+)
+infere_image_uri = (
+    "763104351884.dkr.ecr.us-east-1.amazonaws.com/autogluon-inference:0.4.2-cpu-py38-ubuntu20.04"
+)
```

Comment:
Can these be fetched programmatically (such as `sagemaker.image_uris.retrieve()`) with the current AWS region, rather than hard-coding us-east-1?

Reply:
Thanks for recommendation. implemented the suggested approach